chain/consensus/mir/mempool/handlers/batches.go

```diff
@@ -40,10 +40,20 @@ func IncludeBatchCreation(
 	})
 
 	mpdsl.UponRequestBatch(m, func(origin *mempoolpb.RequestBatchOrigin) error {
-		var txIDs []t.TxID
 		var txs []*requestpb.Request
+		var txIDs []t.TxID
 		batchSize := 0
 
+		submitChan := make(chan []*requestpb.Request)
+
+		state.DescriptorChan <- types.Descriptor{
+			Limit:      0,
+			SubmitChan: submitChan,
+		}
+
+		receivedRequests := <-submitChan
+		mpdsl.RequestTransactionIDs(m, mc.Self, receivedRequests, &requestTxIDsContext{receivedRequests})
```

Comment:
Ok I was probably wrong about only having to change this function. I didn't realize that the transaction IDs still must be computed. This implementation only gets the transactions from outside and has their IDs computed, but does not include them in the batch that is emitted as an event. In a strange way it probably works, just delayed, but this is not what I had in mind. What I meant is to directly emit a batch with the transactions received through the channel, without any buffering. For that, there needs to be more change to the module implementation.

Reply:
It seems to me this is a [feature](https://github.com/filecoin-project/mir/blob/8e480ade52d4d62cc762682cfbd6c2a0e25d4c77/pkg/availability/multisigcollector/internal/parts/batchreconstruction/batchreconstruction.go#L96.).

I can delete calling of `mpdsl.RequestTransactionIDs` and implement hashing without sending an event but it will be also called later in batch reconstruction that is a not part of mempool.

Let's discuss in sync